File: isu-service/src/main/java/ru/itmo/soa/isu/StudyGroupsClient.java
package ru.itmo.soa.isu;

import ru.itmo.soa.model.ApiModels.StudyGroup;
import ru.itmo.soa.model.ApiModels.StudyGroupRequest;

public interface StudyGroupsClient {
    StudyGroup get(int id);
    StudyGroup update(int id, StudyGroupRequest request);
    void delete(int id);
}

File: src/main/kotlin/ru/itmo/soa/isu/StudyGroupsClient.kt
package ru.itmo.soa.isu

import ru.itmo.soa.api.dto.StudyGroupRequest
import ru.itmo.soa.api.dto.StudyGroupResponse

interface StudyGroupsClient {
    fun get(id: Int): StudyGroupResponse
    fun update(id: Int, request: StudyGroupRequest): StudyGroupResponse
    fun delete(id: Int)
}
